pkg/server/sandbox_reset_windows.go

```diff
@@ -0,0 +1,242 @@
+//go:build windows
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"github.com/containerd/containerd"
+	containerdio "github.com/containerd/containerd/cio"
+	"github.com/containerd/containerd/containers"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/log"
+	"github.com/containerd/containerd/snapshots"
+	customopts "github.com/containerd/cri/pkg/containerd/opts"
+	ctrdutil "github.com/containerd/cri/pkg/containerd/util"
+	"github.com/containerd/cri/pkg/netns"
+	"github.com/containerd/cri/pkg/store"
+	sandboxstore "github.com/containerd/cri/pkg/store/sandbox"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+// ResetPodSandbox resets resources and state of the sandbox and its associated containers
+func (c *criService) resetSandbox(ctx context.Context, sandbox sandboxstore.Sandbox) (retErr error) {
+	var (
+		id     = sandbox.ID
+		entity = log.G(ctx).WithField("sandboxID", id)
+	)
+
+	// Return error if sandbox is in running in unknown state
+	state := sandbox.Status.Get().State
+	switch state {
+	case sandboxstore.StateReady:
+		entity.Debugf("sandbox is already running")
+		return nil
+	case sandboxstore.StateNotReady:
+	default:
+		return errors.Errorf("sandbox %q is in invalid state %q", id, state)
+	}
+
+	netnsPath := sandbox.NetNS.GetPath()
+	entity.WithField("netNSPath", netnsPath).Debugf("recreating HCN network namespace")
+	// If containerd was closed while sandbox was running, HCN namespace may not have been
+	// properly closed. Loading that namespace will cause `task.New` to error if the sandbox
+	// task endpoint already exists and is attached to the switch.
+	// So remove the namespace and then recreate it
+	nns, err := netns.NewNetNSWithPath(netnsPath, true /*removeExisting*/)
+	if err != nil {
+		return errors.Wrapf(err, "failed to recreate network namespace for sandbox %q", id)
+	}
+	if nns.GetPath() != netnsPath {
+		return errors.Errorf("recreated network namespace for sandbox %q is %q, not %q", id, nns.GetPath(), netnsPath)
+	}
+
+	defer func() {
+		if retErr != nil {
+			if err := sandbox.NetNS.Remove(); err != nil {
+				entity.WithError(err).WithField("netNS", sandbox.NetNSPath).
+					Error("Failed to remove network namespace")
+			}
+			sandbox.NetNSPath = ""
```

Comment:
What's the purpose of this line?

Reply:
I think it resets the network namespace path, but I am not sure. I copied it from [`RunPodSandbox`](https://github.com/kevpar/cri/blob/9b6416bd48f6507d49939e2d0b9af8c93f442909/pkg/server/sandbox_run_unix.go#L121), but I dont think its actually doing anything here since the Sandbox metadata should be immutable at this point, if the sandbox is added to the sandbox store.
Ill remove it